app/controllers/application_controller.rb

```diff
@@ -2,9 +2,156 @@
 
 class ApplicationController < Sinatra::Base
 
+  helpers SessionHelpers
+  helpers UserHelpers
+  helpers TweetHelpers
+
+  enable :sessions
+  set :session_secret, "secret"
+  use Rack::Flash
+
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    redirect to "/tweets" if is_logged_in?
+
+    erb :'/users/create_user'
+  end
+
+  get '/login' do
+    redirect to "/tweets" if is_logged_in?
+
+    erb :'users/login'
+  end
+
+  get '/logout' do
+    session.clear if is_logged_in?
+    redirect to "/login"
+  end
+
+  get '/tweets/new' do
+    if is_logged_in?
+      erb :"/tweets/create_tweet"
+    else
+      flash[:notice] = "You must new logged in to tweet!"
+      redirect to "/login"
+    end
+  end
+
+  post '/tweets' do
+    if is_logged_in?
+      if validate_new_tweet_params(params)
+        tweet = Tweet.create(content: params[:content],
```

Comment:
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.